implants/lib/pb/src/xchacha.rs

```diff
@@ -0,0 +1,223 @@
+use anyhow::{Context, Result};
+use bytes::{Buf, BufMut};
+use chacha20poly1305::{aead::generic_array::GenericArray, aead::Aead, AeadCore, KeyInit};
+use const_decoder::Decoder as const_decode;
+use prost::Message;
+use rand::rngs::OsRng;
+use rand_chacha::rand_core::SeedableRng;
+use std::{
+    collections::HashMap,
+    io::{Read, Write},
+    marker::PhantomData,
+    sync::{Mutex, OnceLock},
+};
+use tonic::{
+    codec::{Codec, DecodeBuf, Decoder, EncodeBuf, Encoder},
+    Status,
+};
+use x25519_dalek::{EphemeralSecret, PublicKey};
+
+const SERVER_PUBKEY_STR: &str = env!("IMIX_SERVER_PUBKEY");
+const SERVER_PUBKEY: [u8; 32] = const_decode::Base64.decode(SERVER_PUBKEY_STR.as_bytes());
+
```

Comment:
Move these into the agent config section so all env vars are centrally managed.

Reply:
Ran into issues trying to do this so they'll live here for now.
Likely need to reslove some tonic codegen nonsense first.